shared/src/commonMain/kotlin/com/devil/phoenixproject/presentation/manager/DefaultWorkoutSessionManager.kt

```
@@ -258,6 +265,19 @@ class DefaultWorkoutSessionManager(
             override fun proceedFromSummary() = this@DefaultWorkoutSessionManager.proceedFromSummary()
         }
 
+        scope.launch {
+            try {
+                preferencesManager.preferencesFlow.collect { prefs ->
+                    coordinator.updateVbtSettings(
+                        velocityLossThresholdPercent = prefs.velocityLossThresholdPercent.toFloat(),
+                        autoEndOnVelocityLoss = prefs.autoEndOnVelocityLoss,
+                    )
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)